packages/survey/tests/Question/ItemGroup.test.tsx

```diff
@@ -0,0 +1,124 @@
+  withLabels: false,
+  withImages: false,
+}
+
+describe('ItemGroup component', () => {
+  it('passes a11y test', async () => {
+    await testA11y(<ItemGroup {...mockDefaultProps} />)
+  })
+
+  it('renders correctly as checkbox', () => {
+    const { getByTestId } = render(<ItemGroup {...mockDefaultProps} />)
+    expect(getByTestId('ItemGroup')).toBeTruthy()
+  })
+
+  it('renders correctly as checkbox', () => {
+    const { getByTestId } = render(<ItemGroup {...mockDefaultProps} itemType={'radio'} />)
+    expect(getByTestId('ItemGroup')).toBeTruthy()
+  })
+
+  it('renders correctly with labels', () => {
+    const { getByTestId } = render(<ItemGroup {...mockDefaultProps} withLabels={true} />)
+    expect(getByTestId('ItemGroup')).toBeTruthy()
+  })
+
+  it('renders correctly horizontally', () => {
+    const { getByTestId } = render(<ItemGroup {...mockDefaultProps} direction={'horizontal'} />)
+    expect(getByTestId('ItemGroup')).toBeTruthy()
+  })
+
+  it('renders with images', () => {
+    const { getByTestId } = render(<ItemGroup {...mockDefaultProps} withImages={true} />)
+    expect(getByTestId('ItemGroup')).toBeTruthy()
+  })
+
+  it('renders with images and with labels', () => {
+    const { getByTestId } = render(
+      <ItemGroup {...mockDefaultProps} withImages={true} withLabels={true} />
+    )
+    expect(getByTestId('ItemGroup')).toBeTruthy()
+  })
+
+  it('renders with images, horizontal and max columns', () => {
+    const { getByTestId } = render(
+      <ItemGroup {...mockDefaultProps} withImages={true} direction={'horizontal'} maxColumns={3} />
+    )
+    expect(getByTestId('ItemGroup')).toBeTruthy()
+  })
+
+  it('renders with light active color', () => {
+    const { getByTestId } = render(<ItemGroup {...mockDefaultProps} activeColor={'#75e0f6'} />)
+    expect(getByTestId('ItemGroup')).toBeTruthy()
+  })
+
+  it('can be checked as checkbox', () => {
+    const clickEvent = jest.fn()
+    const { container } = render(<ItemGroup {...mockDefaultProps} onChange={clickEvent} />)
+
+    const val = mockDefaultProps.options[1].value
+    const item = container.querySelectorAll(`input[value="${val}"]`)
+
+    if (item.length > 0) {
+      fireEvent.click(item[0])
+      expect(clickEvent.mock.calls.length).toBe(1)
+    }
+
+    const val2 = mockDefaultProps.options[2].value
+    const item2 = container.querySelectorAll(`input[value="${val}"]`)
+
+    if (val2.length > 0) {
+      fireEvent.click(item2[0])
+      expect(clickEvent.mock.calls.length).toBe(2)
+    }
+  })
+
+  it('can be checked as radio', () => {
+    const clickEvent = jest.fn()
+    const { container } = render(
+      <ItemGroup {...mockDefaultProps} itemType={'radio'} onChange={clickEvent} />
+    )
+
+    const val = mockDefaultProps.options[1].value
+    const item = container.querySelectorAll(`input[value="${val}"]`)
+
+    if (item.length > 0) {
+      fireEvent.click(item[0])
+      expect(clickEvent.mock.calls.length).toBe(1)
+    }
+  })
+})
```

Comment:
From my experience it is easier to maintain components in the long run if each component is tested individually. So in this case I would write the `Item` first and then the `ItemGroup` component test cases.

I mean basically always from the child to the parent. This often makes it easier later to add more components that also include this child component etc.

Reply:
Item tests are also added, since some chakra props are needed and served from ItemGroup I got some troubles to test all...
Check them out if it's a  good way to start!